broker/ga4gh/broker/endpoints/repositories.py

```diff
@@ -0,0 +1,287 @@
+import logging
+import string
+from random import choice
+from typing import (Dict)
+
+from flask import (current_app)
+from pymongo.errors import DuplicateKeyError
+from werkzeug.exceptions import Unauthorized
+
+from broker.errors.exceptions import (
+    RepositoryNotFound,
+    URLNotFound,
+    InternalServerError
+)
+
+logger = logging.getLogger(__name__)
+
+
+def register_repository(data: Dict):
+    """Register a new repository.
+
+    Args:
+         data: Dictionary element of request received.
+
+    Returns:
+        repository_object: Dictionary element of Repository created (
+        identifier and access_token).
+
+    Description:
+        - Takes request data.
+        - Creates a repository object using url in request data.
+        - Generate a unique identifier for the object. (Retries 3 times for
+        generating unique identifier.)
+        - Generates access_token of 32 characters. (access_token will be
+        used later for operations like creating a new build, updating build
+        completion.)
+        - Inserts object in mongodb.
+        - Remove url from the repository_object
+        - Returns id and access_token for the repository_object created.
+
+    Raises:
+        URLNotFound: Raised when request data does not have url string.
+        InternalServerError: Raised when it's unable to create unique
+        identifier for the repository_object.
+
+
+    """
+    retries = 3
+    repository_object = {}
+    try:
+        repository_object['url'] = data['url']
```

Comment:
Isn't this automatically validated by Connexion if the `url` property is required in the corresponding OpenAPI schema?